### ISSUE TITLE
feat: lock wallet to specific dApp

### ISSUE THREAD

**Issue by lutr0** [RabbyHub/Rabby#2523]
Some dApps (e.g. HyperLiquid, dyxd) have the user sign a message to connect a given wallet.

I would like for this specific wallet to remain connected to the Dapp even if at the same time I use different wallets on other dApps.

Alternatively, being able to use at the same time different wallets for different chains would also solve the same issue on my side

**Comment by liqlootdev** [comment]
Yes this is desperately needed, if you work with even just 3-4 wallets daily its such a massive pain in the ass to use Rabby.
